### PR TITLE
EVA-1990 - Support variants with multiple alt and missing GT

### PR DESCRIPTION
**Pull request by sundarvenkata-EBI**

### CODE REVIEW

**Review by tcezard — APPROVED** [review]

File: biodata-models/src/test/java/org/opencb/biodata/models/variant/stats/VariantStatsTest.java [hunk 1]
```diff
@@ -133,5 +135,27 @@ public void testCalculateMultiallelicStats() {
...
+        source.setSamples(sampleNames);
+        String line = "1\t10040\trs123\tT\tA,GC\t.\tPASS\t.\tGT:GL\t"
+                + "2/-1:1,2,3,4,5,6,7,8,9,10";
+
+        List<Variant> result = new VariantVcfFactory().create(source, line);
+        assertEquals(2, result.size());
+
+        Variant variant_C = result.get(0);
+        VariantSourceEntry sourceEntry_C = variant_C.getSourceEntry(source.getFileId(), source.getStudyId());
+        VariantStats multiallelicStats_C = new VariantStats(result.get(0)).calculate(sourceEntry_C.getSamplesData(), sourceEntry_C.getAttributes(), null);
```
Comment (andresfsilva) [inline biodata-models/src/test/java/org/opencb/biodata/models/variant/stats/VariantStatsTest.java]:
```suggestion
        VariantStats multiallelicStats_C = new VariantStats(variant_C).calculate(sourceEntry_C.getSamplesData(), sourceEntry_C.getAttributes(), null);
```

Comment (andresfsilva) [reply biodata-models/src/test/java/org/opencb/biodata/models/variant/stats/VariantStatsTest.java]: Do a line break so it is not that long

File: biodata-models/src/main/java/org/opencb/biodata/models/variant/stats/VariantStats.java [hunk 2]
```diff
@@ -426,15 +426,23 @@ public VariantStats calculate(Map<String, Map<String, String>> samplesData, Map<
                     if (g.getAllele(0) < 0) {
                         this.setMissingAlleles(this.getMissingAlleles() + 1);
                     } else {
-                        allelesCount[g.getAllele(0)]++;
-                        totalAllelesCount++;
+                        //Allele counts for multiple alternates not supported by biodata!
+                        //Skip allele counts for genotypes like 2/-1 (getAllele(0) returns 2)
+                        if (g.getAllele(0) <= 1) {
```
Comment (andresfsilva) [inline biodata-models/src/main/java/org/opencb/biodata/models/variant/stats/VariantStats.java]: It means the same but I find this clearer. If you change `g.getAllele(0)` to something like `firstAllele` even better
```suggestion
                        if (g.getAllele(0) == 0 || g.getAllele(0) == 1) {
```

**Review by andresfsilva — COMMENTED** [review]
It looks good overall to solve the problem we are getting but I am wondering if in the long term we should support multiple alternates so we do more accurate counts. Although we will probably move from this library in the future.

**Review by andresfsilva — APPROVED** [review]

File: biodata-models/src/main/java/org/opencb/biodata/models/variant/stats/VariantStats.java (hunk 2 above)
Comment (sundarvenkata-EBI) [reply biodata-models/src/main/java/org/opencb/biodata/models/variant/stats/VariantStats.java]: Fair point. Addressed!

File: biodata-models/src/test/java/org/opencb/biodata/models/variant/stats/VariantStatsTest.java (hunk 1 above)
Comment (sundarvenkata-EBI) [reply biodata-models/src/test/java/org/opencb/biodata/models/variant/stats/VariantStatsTest.java]: Done..

**Comment by sundarvenkata-EBI** [comment]
> It looks good overall to solve the problem we are getting but I am wondering if in the long term we should support multiple alternates so we do more accurate counts. Although we will probably move from this library in the future.

@andresfsilva Agree. I have already created a [ticket](https://www.ebi.ac.uk/panda/jira/browse/EVA-2285) to look at this in the future.

**Pull request merged**